eZ/Publish/Core/Repository/Mapper/ContentMapper.php

```diff
@@ -227,4 +271,80 @@ private function cloneField(Field $field, array $overrides = []): Field
 
         return new Field($fieldData);
     }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\Field[] $updatedFields
+     *
+     * @return \eZ\Publish\API\Repository\Values\Content\Field[]
+     */
+    public function getFieldsForUpdate(array $updatedFields, Content $content): array
+    {
+        $contentType = $content->getContentType();
+        $fields = [];
+
+        foreach ($updatedFields as $updatedField) {
+            $fieldDefinition = $contentType->getFieldDefinition($updatedField->fieldDefIdentifier);
+
+            if ($fieldDefinition === null) {
+                throw new ContentValidationException(
+                    "Field definition '%identifier%' does not exist in given Content Type",
+                    ['%identifier%' => $updatedField->fieldDefIdentifier]
+                );
+            }
+
+            $fieldType = $this->fieldTypeRegistry->getFieldType($fieldDefinition->fieldTypeIdentifier);
+
+            $field = $content->getField($updatedField->fieldDefIdentifier);
+            $updatedFieldValue = $this->getFieldValueForUpdate(
+                $updatedField,
+                $field,
+                $contentType->getFieldDefinition($updatedField->fieldDefIdentifier),
+                !in_array($updatedField->languageCode, $content->versionInfo->languageCodes)
+            );
+
+            if (!empty($field)) {
+                $updatedFieldHash = md5(serialize($fieldType->toHash($updatedFieldValue)));
+                $contentFieldHash = md5(serialize($fieldType->toHash($field->value)));
```

Comment:
can we avoid `serialize` in favor of e.g. `json_encode` or something else? `serialize` proved to have security vulnerabilities in the past.